src/utils/variable/regexVariable.js

```diff
@@ -1,4 +1,4 @@
-const def = '\\${([a-zA-Z0-9_\\.[\\]]*)}'
+const def = '\\${(([a-zA-Z0-9_\\.[\\]]*)[\\s\\|\\|\\s]*([a-zA-Z0-9_\\.\\\'\\"]*))}'
```

Comment:
Since this is already just a string (as opposed to a regex literal `/foobar/`), I think you should split this up and add comments.
```suggestion
const def = '\\${' + // opening ${
  '(([a-zA-Z0-9_\\.[\\]]*)' + // match number of any alpha numeric, ., [, and ]
  '[\\s\\|\\|\\s]*' + // match ' || ' 
  '([a-zA-Z0-9_\\.\\\'\\"]*))' +  // match number of any alpha numeric, ., ", and '
  '}' // closing }
```

In fact.. breaking it up and adding comments forced me to review it... and i'm not sure about `[\\s\\|\\|\\s]*`. The comment I added is what I _think_ you wanted, but I think the regex for that would be `(\s\\|\\|\s)`, where as what you have would match any combination of whitespace and `|`. IE, it would match ` || ` as well as `| |     | ||||   || ` which I don't think is right.

(I wish JS just had [python-style verbose regexes](https://docs.python.org/2/library/re.html#re.VERBOSE))

Reply:
infact maybe the regex for ` || ` should be `(\s+\\|\\|\s+)` (the plus makes the whitespace optional, matching javascript semantics.

Reply:
Thanks for the in-depth review on this on @dschep 👍